pkg/controller/servicebinding/secret_test.go

```diff
@@ -37,18 +35,96 @@ func TestSecretNew(t *testing.T) {
 	s := newSecret(
 		f.FakeDynClient(),
 		ns,
-		name,
+		secretName,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName1, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName1, secretName)
+
+		hashedName2, err := name(hashedName1, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName2, hashedName1)
+	})
+
+	t.Run("check secret creation", func(t *testing.T) {
+		u, err := s.overwrite()
+		assert.NoError(t, err)
+		assertSecretNamespacedName(t, u, ns, secretName)
+		existingSecret, err := s.get()
 		assert.NoError(t, err)
-		assertSecretNamespacedName(t, u, ns, name)
+		hashedName, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, existingSecret.GetName())
 	})
 
-	t.Run("Get", func(t *testing.T) {
-		u, err := s.get()
+	t.Run("different_secret_payload_are_different_secret", func(t *testing.T) {
+		payload1 := map[string][]byte{"dbName": []byte("postgres")}
+		value1 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			secretName,
+			payload1,
+			secretOwnerReference,
+		)
+		secret1, err := value1.overwrite()
+		assert.NoError(t, err)
+
+		payload2 := map[string][]byte{"user": []byte("admin")}
+		value2 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			secretName,
+			payload2,
+			secretOwnerReference,
+		)
+		secret2, err := value2.overwrite()
+		assert.NoError(t, err)
+
+		assert.NotEqual(t, secret1.GetName(), secret2.GetName())
+	})
+
+	t.Run("secret_with_empty_payload", func(t *testing.T) {
+		value1 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			secretName,
+			nil,
+			secretOwnerReference,
+		)
+		secret1, err := value1.overwrite()
+		assert.NoError(t, err)
+		assert.NotEqual(t, secretName, secret1.GetName())
+	})
+
+	t.Run("get_existing_secret", func(t *testing.T) {
```

Comment:
the test name suggest that we test `get()` function, but we do not. Please describe better what is the scenario we test here.